lib/cn/kblock_builder.c

```diff
@@ -1152,6 +1162,7 @@ kbb_add_entry(
         return err;
     if (added) {
         hlog_add(bld->hlog, hash);
+        hlog_add(bld->curr.hlog, hash);
```

Comment:
you could rename `bld->hlog` to `bld->kvset_hlog` to make it more self documenting.